lib/tapjoy/ldap/api/group.rb

```diff
@@ -0,0 +1,72 @@
+require 'memoist'
+module Tapjoy
+  module LDAP
+    module API
+      module Group
+        class << self
+          extend Memoist
+          def create(group_name, group_type)
+            Tapjoy::LDAP::client.add(
+              distinguished_name(group_name),
+              ldap_attr(group_name, group_type)
+            )
+          end
+
+          def destroy(group_name)
+            Tapjoy::LDAP::client.delete(distinguished_name(group_name))
+          end
+
+          def update(group_name, username, operation)
+            Tapjoy::LDAP::client.modify(
+              distinguished_name(group_name),
+              [[operation, :memberUid, username]]
+            )
+          end
+
+          def index
+            Tapjoy::LDAP::client.search('*', filter(objectClass: 'posixGroup'))
+          end
+
+          # Lookup GID for the given group
+          def lookup_id(groupname)
+            gidnumber = []
+
+            oc_filter = Net::LDAP::Filter.eq('objectclass', 'posixGroup')
+            cn_filter = Net::LDAP::Filter.eq('cn', groupname)
+            filter    = Net::LDAP::Filter.join(oc_filter, cn_filter)
+
+            results = Tapjoy::LDAP::client.search(['gidNumber'], filter)
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Do not use `::` for method calls.](https://www.codacy.com/app/ali-tayarani/ldap_tools/file/2633306012/issues/source?bid=3316970&fileBranchId=3318857#l38)